### ISSUE TITLE
Keep outputting '0it [00:00, ?it/s]'

### ISSUE THREAD

**Issue by tingkai-zhang** [IndicoDataSolutions/finetune#67]
**Describe the bug**
I use the following code to run a demo on SNLI dataset.
It keeps outputting '0it [00:00, ?it/s]'

The output file looks like this:
```
 FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
```

**Minimum Reproducible Example**
```
def trim(string):
    try:
        string = ' '.join(string.split(' ')[:256]).rstrip('\n')
        return string       
    except:
        raise ValueError(f'{string}')

def read_file(file):
    with open(file) as f:
        lines=[]
        for line in f:
            line = trim(line)
            lines.append(line)
    return lines

if __name__ == "__main__":

    trainX1 = read_file('premise_snli_1.0_train.txt')
    trainX2 = read_file('hypothesis_snli_1.0_train.txt')
    trainY = read_file('label_snli_1.0_train.txt')

    testX1 = read_file('premise_snli_1.0_test.txt')
    testX2 = read_file('hypothesis_snli_1.0_test.txt')
    testY = read_file('label_snli_1.0_test.txt')

    model = Entailment(verbose=True)
    model.fit(trainX1, trainX2, trainY)
    model.save('./saved_snli_model')
    pred_result = model.predict(testX1, testX2)
```
`premise_snli_1.0_train.txt` is a file where each line is a sentence.
In config.py file i set the max_length to be 258, batch_size to be 8

**Comment by benleetownsend** [comment]
TLDR; If it seems like the model isn't training, just be patient, at this point it hasn't started. I am fixing this in #62 to give a clearer progress bar.

Can I confirm that after these lines have completed, the model trains and functions as expected?

Due to this model being designed to work on small datasets, all the tokenization and encoding is done as a single first step before the model is initialized and trained. For this reason, training on a large-ish dataset like SNLI may take a little patience for this preprocessing to complete.

What you are seeing here is a misplaced progress bar that has not been noticed as on smaller datasets the tokenization process completes very quickly.

I am including a fix with a clearer progress bar in PR: #62.

Thank you for highlighting this issue and your ongoing interest in the project.

**Comment by benleetownsend** [comment]
One other thing to note, is that the max length is in tokens, not in characters like your code is enforcing. This max length is handled internally, and any trimming is done automatically, the warning that is raised simply means that information is being lost, but if you are trimming anyway. Its better to feed the whole sentences and allow the model to maintain as much information as possible.

**Comment by tingkai-zhang** [comment]
Hi, thanks for replying! Awesome work you done! I will keep an eye on this project.

Here is what i did:
1. I run the demo code without the trimming part. It keeps outputting  `0it [00:00, ?it/s]` while giving me the warning like this: 
`Some examples are longer than the max_length. Please trim documents or increase `max_length`.
Fallback behaviour is to use the first 254 byte-pair encoded tokens`

2. I stopped it before it started training. 

3. Then I set the max_length to be 258 in config.py, trim the sentence to be 256 words.

4. It still output `0it [00:00, ?it/s]` without giving me the warning like before. But it started training after these lines are completed.

Since the sentences will be trimmed anyway, I felt it doesn't hurt if i trim the sentence outside the model and this solution works.

The progress bar shows that the time iterates over SNLI (about 59w sentences) is 27 hours(I am using 2 * 1080TI). I set the batch size to be 8 with other parameters untouched. My colleges revised the code OpenAI released and he run his own code for 8 hours in total on SNLI. 
I was wondering why there is huge time difference between his code and this one.
May i ask the training time you run the code on a similar size dataset or give me a time as reference?
Thank you!

**Comment by madisonmay** [comment]
Closing this issue as the original issue with unnecessary TQDM logs has now been resolved on the master branch. Thanks for the bug report, feel free to open another issue if you have other Qs we might be able to help out with!
